src/Npgsql.GeoJSON/NpgsqlGeoJSONExtensions.cs

```diff
@@ -17,7 +18,20 @@ public static class NpgsqlGeoJSONExtensions
     /// <param name="geographyAsDefault">Specifies that the geography type is used for mapping by default.</param>
     public static INpgsqlTypeMapper UseGeoJson(this INpgsqlTypeMapper mapper, GeoJSONOptions options = GeoJSONOptions.None, bool geographyAsDefault = false)
     {
-        mapper.AddTypeResolverFactory(new GeoJSONTypeHandlerResolverFactory(options, geographyAsDefault));
+        mapper.AddTypeInfoResolver(new GeoJSONTypeInfoResolver(options, geographyAsDefault, crsMap: null));
         return mapper;
     }
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Sets up GeoJSON mappings for the PostGIS types.
+    /// </summary>
+    /// <param name="mapper">The type mapper to set up (global or connection-specific)</param>
+    /// <param name="crsMap">A custom crs map that might contain more or less entries than the default well-known crs map.</param>
+    /// <param name="options">Options to use when constructing objects.</param>
+    /// <param name="geographyAsDefault">Specifies that the geography type is used for mapping by default.</param>
+    public static INpgsqlTypeMapper UseGeoJson(this INpgsqlTypeMapper mapper, CrsMap crsMap, GeoJSONOptions options = GeoJSONOptions.None, bool geographyAsDefault = false)
```

Comment:
I like this solution... Some day we can think about allowing plugins to register "loading logic" on their data source, but if GeoJSON is the only example of this, meh.